web/controllers/donation_goal_controller.ex

```diff
@@ -0,0 +1,24 @@
+defmodule CodeCorps.DonationGoalController do
+  use CodeCorps.Web, :controller
+  use JaResource
+
+  alias CodeCorps.DonationGoal
+
+  plug :load_and_authorize_changeset, model: DonationGoal, only: [:create]
+  plug :load_and_authorize_resource, model: DonationGoal, only: [:update, :delete]
+  plug JaResource
+
+  def handle_create(conn, attributes) do
+    %DonationGoal{}
+    |> DonationGoal.create_changeset(attributes)
+    |> Repo.insert
+    |> CodeCorps.Analytics.Segment.track(:created, conn)
+  end
+
+  def handle_update(conn, record, attributes) do
+    record
+    |> DonationGoal.update_changeset(attributes)
+    |> Repo.update
+    |> CodeCorps.Analytics.Segment.track(:edited, conn)
```

Comment:
Actually, this is a copy-paste remnant. I did not implement tracking for donation goals and I'm not sure I should as part of this PR. We can decide how and what to track later.